test/python/utils/mitigation/test_meas.py

```diff
@@ -0,0 +1,707 @@
+                }
+            )
+
+        # Set the state labels
+        state_labels = ["000", "001", "010", "011", "100", "101", "110", "111"]
+        meas_cal = CompleteMeasFitter(None, state_labels, circlabel="test")
+
+        for tst_index, _ in enumerate(tests):
+            # Set the calibration matrix
+            meas_cal.cal_matrix = tests[tst_index]["cal_matrix"]
+            # Calculate the fidelity
+            fidelity = meas_cal.readout_fidelity()
+
+            meas_filter = MeasurementFilter(tests[tst_index]["cal_matrix"], state_labels)
+
+            # Calculate the results after mitigation
+            output_results_pseudo_inverse = meas_filter.apply(
+                tests[tst_index]["results"], method="pseudo_inverse"
+            )
+            output_results_least_square = meas_filter.apply(
+                tests[tst_index]["results"], method="least_squares"
+            )
+
+            # Compare with expected fidelity and expected results
+            self.assertAlmostEqual(fidelity, tests[tst_index]["fidelity"], places=0)
+            self.assertAlmostEqual(
+                output_results_pseudo_inverse["000"],
+                tests[tst_index]["results_pseudo_inverse"]["000"],
+                places=0,
+            )
+
+            self.assertAlmostEqual(
+                output_results_least_square["000"],
+                tests[tst_index]["results_least_square"]["000"],
+                places=0,
+            )
+
+            self.assertAlmostEqual(
+                output_results_pseudo_inverse["111"],
+                tests[tst_index]["results_pseudo_inverse"]["111"],
+                places=0,
+            )
+
+            self.assertAlmostEqual(
+                output_results_least_square["111"],
+                tests[tst_index]["results_least_square"]["111"],
+                places=0,
+            )
+
+    def test_tensored_meas_fitter_with_noise(self):
+        """Test the TensoredFitter with noise."""
+        cal_results, mit_pattern, circuit_results, meas_layout = tensored_calib_circ_execution(
+            1000, SEED
+        )
+
+        meas_cal = TensoredMeasFitter(cal_results, mit_pattern=mit_pattern)
+        meas_filter = meas_cal.filter
+
+        # Calculate the results after mitigation
+        results_pseudo_inverse = meas_filter.apply(
+            circuit_results.get_counts(), method="pseudo_inverse", meas_layout=meas_layout
+        )
+        results_least_square = meas_filter.apply(
+            circuit_results.get_counts(), method="least_squares", meas_layout=meas_layout
+        )
+        saved_info = {
+            "cal_results": cal_results.to_dict(),
+            "results": circuit_results.to_dict(),
+            "mit_pattern": mit_pattern,
+            "meas_layout": meas_layout,
+            "fidelity": meas_cal.readout_fidelity(),
+            "results_pseudo_inverse": results_pseudo_inverse,
+            "results_least_square": results_least_square,
+        }
```

Comment:
Again, in Ignis this test loads from a known file, but generating the results like this takes well under a second, so I think we're fine.